crates/primitives/src/l1.rs

```diff
@@ -983,6 +983,113 @@ impl<'a> Arbitrary<'a> for TaprootSpendPath {
     }
 }
 
+/// [Borsh](borsh)-friendly Bitcoin [`Transaction`].
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BitcoinTx(Transaction);
+
+impl From<Transaction> for BitcoinTx {
+    fn from(value: Transaction) -> Self {
+        Self(value)
+    }
+}
+
+impl From<BitcoinTx> for Transaction {
+    fn from(value: BitcoinTx) -> Self {
+        value.0
+    }
+}
+
+/// Implement BorshSerialize using Bitcoin consensus serialization.
+impl BorshSerialize for BitcoinTx {
+    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        // Use bitcoin's consensus serialization
+        let tx_bytes = serialize(&self.0);
+        BorshSerialize::serialize(&(tx_bytes.len() as u32), writer)?;
+        writer.write_all(&tx_bytes)
+    }
+}
+
+/// Implement BorshDeserialize using Bitcoin consensus deserialization.
+impl BorshDeserialize for BitcoinTx {
+    fn deserialize_reader<R: Read>(reader: &mut R) -> std::io::Result<Self> {
+        // First, read a Vec<u8> using Borsh (this picks up the length)
+        let tx_len = u32::deserialize_reader(reader)? as usize;
+        let mut tx_bytes = vec![0u8; tx_len];
+        reader.read_exact(&mut tx_bytes)?;
+
+        // Now parse those bytes with bitcoin consensus
+        let tx = deserialize(&tx_bytes).map_err(|err| Error::new(ErrorKind::InvalidData, err))?;
+
+        Ok(BitcoinTx(tx))
+    }
+}
+
+impl<'a> arbitrary::Arbitrary<'a> for BitcoinTx {
```

Comment:
This could be good to keep on a `RawBitcoinTx` type if it's a wrapper around `Vec<u8>`.